woocommerce/woocommerce
# Ensure HPOS is disabled if ENABLE_HPOS is set to '0'

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Currently, when setting up an environment using wp-env, HPOS is being set by default. We want to be able to test the environment when HPOS is not set or set to 0 and, for some reason, after login, HPOS is being set to on, even if prior to login `woocommerce_custom_orders_table_enabled` has been set to `no`. This ensures the appropriate setting is set via the global-setup.js after login.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. CI tests pass
2. CI tests pass and `HPOS Switched off successfully` is logged when ENABLE_HPOS env variable is set to '0' (check [Run tests against PR in an environment with COT disabled / Runs E2E tests with COT disabled](https://github.com/woocommerce/woocommerce/actions/runs/6399348552/job/17371243661?pr=40528))
3. 
![image](https://github.com/woocommerce/woocommerce/assets/105309450/f4420d30-91f4-4554-b618-a31549c13186)

4.
5.  CI tests pass and `HPOS Switched off successfully` is NOT logged when ENABLE_HPOS env variable is not set (check [Run tests against PR / Runs E2E tests](https://github.com/woocommerce/woocommerce/actions/runs/6399348557/job/17371243555?pr=40528))
6. Locally, setup env with `pnpm test:env`
7. navigate to `plugins/woocommerce` with `cd plugins/woocommerce`
8. execute any e2e test such as `pnpm test:e2e-pw ./tests/e2e-pw/tests/activate-and-setup/basic-setup.spec.js` and do not set ENABLE_HPOS env variable - `HPOS Switched off successfully` and `HPOS Switched on successfully` should NOT be logged - navigate to `http://localhost:8086/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and HPOS should be enabled
9. execute any api test such as `pnpm test:api-pw ./tests/api-core-tests/tests/hello/hello.test.js` -  `HPOS Switched off successfully` and `HPOS Switched on successfully` should NOT be logged - navigate to `http://localhost:8086/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and HPOS should be enabled
10. set ENABLE_HPOS env variable to 0 with `export ENABLE_HPOS=0` and execute any e2e test such as `pnpm test:e2e-pw ./tests/e2e-pw/tests/activate-and-setup/basic-setup.spec.js` - `HPOS Switched off successfully` should be logged - navigate to `http://localhost:8086/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and HPOS should be disabled
11. execute any api test such as `pnpm test:api-pw ./tests/api-core-tests/tests/hello/hello.test.js` - `HPOS Switched off successfully` should be logged - navigate to `http://localhost:8086/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and HPOS should be disabled
12. set ENABLE_HPOS env variable to 1 with `export ENABLE_HPOS=1` and execute any e2e test such as `pnpm test:e2e-pw ./tests/e2e-pw/tests/activate-and-setup/basic-setup.spec.js` - `HPOS Switched on successfully` should be logged - navigate to `http://localhost:8086/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and HPOS should be enabled
13. execute any api test such as `pnpm test:api-pw ./tests/api-core-tests/tests/hello/hello.test.js`  `HPOS Switched on successfully` should be logged - navigate to `http://localhost:8086/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` and HPOS should be enabled



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
